Tests/AppcuesKitTests/Networking/ExperienceDecodeTests.swift

```diff
@@ -0,0 +1,259 @@
+                                }
+                            ],
+                            "actions": {
+                                "1f9d1c11-525f-4f1c-afc1-d91b97c6a7d4": [
+                                    {
+                                        "on": "tap",
+                                        "type": "action-type",
+                                        "config": {
+                                            "actionProp": true
+                                        }
+                                    }
+                                ]
+                            }
+                        }
+                    ]
+                }
+            ]
+        }
+        """#.data(using: .utf8)!
+
+        // Act
+        let experience = try? NetworkClient.decoder.decode(Experience.self, from: data)
+
+        // Assert
+        XCTAssertNotNil(experience)
+    }
+
+    func testDuplicateStepTraitDecode() throws {
+        // Arrange
+        let data = #"""
+        {
+            "id": "05e78601-d7d8-449f-a074-fe3ae1144366",
+            "name": "duplicate step trait",
+            "type": "mobile",
+            "traits": [
+                {
+                    "type": "experience-trait"
+                }
+            ],
+            "steps": [
+                {
+                    "id": "f7edb07c-5f96-4440-9b70-3bdd0b7675b0",
+                    "type": "group",
+                    "actions": {},
+                    "traits": [
+                        {
+                            "type": "group-trait"
+                        }
+                    ],
+                    "children": [
+                        {
+                            "id": "68bb5cc4-99b4-4f5c-a787-930ba921fc05",
+                            "type": "modal",
+                            "content": {
+                                "type": "spacer",
+                                "id": "1f9d1c11-525f-4f1c-afc1-d91b97c6a7d4"
+                            },
+                            "traits": [
+                                {
+                                    "type": "step-trait"
+                                },
+                                {
+                                    "type": "step-trait"
+                                }
+                            ],
+                            "actions": {}
+                        }
+                    ]
+                }
+            ]
+        }
+        """#.data(using: .utf8)!
+
+        // Act
+        var experience: Experience?
+        var error: String?
+        do {
+            experience = try NetworkClient.decoder.decode(Experience.self, from: data)
+        } catch let DecodingError.dataCorrupted(context) {
+            error = "\(context)"
+        }
+
+        // Assert
+        XCTAssertNil(experience)
+        XCTAssertNotNil(error)
+        XCTAssertTrue(error!.contains("multiple traits of same type are not supported: step-trait"))
```

Comment:
```swift
let unwrappedError = try XCTUnwrap(error)
XCTAssertTrue(unwrappedError.contains("multiple traits of same type are not supported: step-trait"))

// or even
XCTAssertTrue(try XCTUnwrap(error).contains("multiple traits of same type are not supported: step-trait"))
```
is a slightly better option since it saves the test from crashing on line 135 if error is actually `nil`. Doesn't matter too much, but it is nicer if the test ever does fail.

Reply:
nice, thanks for the tip on `XCTUnwrap` - updated these